server/server.go

```diff
@@ -101,8 +103,53 @@ type Server struct {
 	logProps *log.ZapProperties
 }
 
+// HandlerBuilder builds a server HTTP handler.
+type HandlerBuilder func(*Server) (http.Handler, APIGroupInfo)
+
+// APIGroupInfo used to regster the api service.
+type APIGroupInfo struct {
+	IsCore  bool
+	Group   string
+	Version string
+}
+
+const (
+	//CorePath  the core group, is at REST path `/pd/api/v1`.
+	CorePath = "/pd/api/v1"
+	// ExtensionsPath the named groups are REST at `/pd/apis/{GROUP_NAME}/{Version}`.
+	ExtensionsPath = "/pd/apis"
```

Comment:
How to understand `api` and `apis`, maybe use `/pd/ext/`, `/pd/extension/` or `/pd/plugin/` ?

Reply:
@shafreeck  I prefer all API follower this style `/pd/apis/{GROUP_NAME}/{Version}`, but we have a history of burdens. the group may not only the extension but also some core components like each scheduler. so do you have a suggestion?